server/src/controllers/orderController.service.ts

```diff
@@ -70,129 +74,151 @@ export const readerWebhook = async (
-        refund.refund_status = state.failed;
-        break;
-    }
-
-    await prisma.refunds.update({
+    return {
       where: {
         id: refund.id,
       },
       data: {
-        refund_status: refund.refund_status,
+        refund_status: getRefundStatus(refundObject.status),
       },
-    });
+    };
   }));
+
+  await prisma.$transaction(queries
+    .filter((query): query is PromiseFulfilledResult<any> => query.status === 'fulfilled')
+    .map((query) => prisma.refunds.update(query.value)));
 };
 
+export const getRefundStatus = (refundStatus?: string) => {
+  switch (refundStatus) {
+  case 'succeeded':
+    return state.completed;
+  case 'pending':
+    return state.in_progress;
+  default:
+    return state.failed;
+  }
+};
+
+
+export interface OrderFulfillmentData {
+  orderStatus: state;
+  orderSource: purchase_source | undefined;
+  orderSubtotal: number;
+  discountTotal: number;
+  feeTotal: number;
+  orderTicketItems?: any[];
+  donationItem?: any;
+  orderSubscriptionItems?: any[];
+  eventInstanceQueries?: any[];
+  paymentIntent?: string;
+  checkoutSession?: string;
+  discountId?: number;
+  contactid?: number;
+}
+
 export const orderFulfillment = async (
-    prisma: ExtendedPrismaClient,
-    eventInstanceQueries: any[],
-    orderSubtotal: number,
-    discountTotal: number,
-    feeTotal: number,
-    orderItems: {
-        orderTicketItems?: any[],
-        donationItem?: any,
-        orderSubscriptionItems?: any[],
-    },
-    checkoutSession?: string,
-    discountId?: number,
-    paymentIntent?: string,
+  prisma: ExtendedPrismaClient,
+  {
+    orderStatus,
+    orderSource,
+    orderSubtotal,
+    discountTotal = 0,
+    feeTotal = 0,
+    donationItem,
+    orderTicketItems = [],
+    orderSubscriptionItems = [],
+    eventInstanceQueries = [],
+    paymentIntent,
+    checkoutSession,
+    discountId,
+    contactid,
+  }: OrderFulfillmentData,
 ) => {
-  const {orderTicketItems, donationItem, orderSubscriptionItems} = orderItems;
   const result = await prisma.$transaction([
     prisma.orders.create({
       data: {
-        checkout_sessions: checkoutSession,
+        contactid_fk: contactid,
         discountid_fk: discountId,
         ordersubtotal: orderSubtotal,
         discounttotal: discountTotal,
         feetotal: feeTotal,
-        ...(orderTicketItems && {orderticketitems: {create: orderTicketItems}}),
-        ...(donationItem && {donation: {create: donationItem}}),
-        ...(orderSubscriptionItems && {subscriptions: {create: orderSubscriptionItems}}),
+        orderticketitems: {create: orderTicketItems},
+        donation: {create: donationItem},
+        subscriptions: {create: orderSubscriptionItems},
+        order_status: orderStatus,
+        order_source: orderSource,
+        checkout_sessions: checkoutSession,
         payment_intent: paymentIntent,
       },
     }),
     ...eventInstanceQueries,
   ]);
+
+  if (orderSubtotal - discountTotal === 0) {
```

Comment:
I think you'd also need to add the `feeTotal` in this calculation.

Reply:
Fixed, thanks for catching that!